long test function names

I am refactoring some code from fruit to pfunit and I have a test function name `populate_mappings_with_multiple_key_value_lines` which generates a line that is longer than 132 characters

```fortran
   if(allocated(t)) deallocate(t)
   allocate(t, source=TestMethod('populate_mappings_with_multiple_key_value_lines', populate_mappings_with_multiple_key_value_lines))
   call suite%addTest(t)
```

The solution is to shorten the name of the function, but is there an easier way to deal with this error or maybe change the code generation to use line continuation? I will create a merge request if you can point me where to put it.

also, Identifying the location of the error message is super vague.

```
Error: Line truncated at (1) [-Werror=line-truncation]
/tests/unit_tests/inputs/fortran_yaml_reader_tests/yaml_file_reader_tests.f90:465:132:

Error: Syntax error in ALLOCATE statement at (1)
f951: some warnings being treated as errors
tests/unit_tests/inputs/fortran_yaml_reader_tests/CMakeFiles/fortran_yaml_test.dir/build.make:78: recipe for target 'tests/unit_tests/inputs/fortran_yaml_reader_tests/CMakeFiles/fortran_yaml_test.dir/yaml_file_reader_tests.F90.o' failed
make[2]: *** [tests/unit_tests/inputs/fortran_yaml_reader_tests/CMakeFiles/fortran_yaml_test.dir/yaml_file_reader_tests.F90.o] Error 1
CMakeFiles/Makefile2:3247: recipe for target 'tests/unit_tests/inputs/fortran_yaml_reader_tests/CMakeFiles/fortran_yaml_test.dir/all' failed
make[1]: *** [tests/unit_tests/inputs/fortran_yaml_reader_tests/CMakeFiles/fortran_yaml_test.dir/all] Error 2
Makefile:165: recipe for target 'all' failed
make: *** [all] Error 2
```

Thank you for your time.

--- a/bin/funit/pFUnitParser.py
+++ b/bin/funit/pFUnitParser.py
@@ -809,7 +809,7 @@ class Parser():
         self.outputFile.write('\nend function ' + self.suiteName + '\n\n')
 
     def addSimpleTestMethod(self, testMethod):
-        args = "'" + testMethod['name'] + "', " + testMethod['name']
+        args = "'" + testMethod['name'] + "', &\n      " + testMethod['name']
         if 'setUp' in testMethod:
             args += ', ' + 'setUp='+testMethod['setUp']
         elif 'setUp' in self.userTestCase:
@@ -833,7 +833,7 @@ class Parser():
 
     def addMpiTestMethod(self, testMethod):
         for npes in testMethod['npRequests']:
-            args = "'" + testMethod['name'] + "', " + testMethod['name'] + ", " + str(npes)
+            args = "'" + testMethod['name'] + "', &\n      " + testMethod['name'] + ", " + str(npes)
             if 'setUp' in testMethod:
                 args += ', ' + testMethod['setUp']
             elif 'setUp' in self.userTestCase:

--- a/tests/funit-core/CMakeLists.txt
+++ b/tests/funit-core/CMakeLists.txt
@@ -43,7 +43,8 @@ set(pf_tests
   Test_TapListener.pf
   Test_NameFilter.pf
   Test_RegularExpression.pf
-  Test_TestMethod_before.pf)
+  Test_TestMethod_before.pf
+  Test_Long_Subroutine_Name.pf)
 
 set (new_test_srcs)
 set (test_srcs

--- a/tests/funit-core/Test_Long_Subroutine_Name.pf
+++ b/tests/funit-core/Test_Long_Subroutine_Name.pf
@@ -1,0 +1,27 @@
+
+module Test_Long_Subroutine_Name
+   use FUnit
+   implicit none
+   
+   @suite(name='AssertLongSubroutine_suite')
+
+contains
+
+   real function add(x, y) result (sum)
+      implicit none
+      real, intent(in) :: x
+      real, intent(in) :: y
+
+      sum = x + y
+
+   end function add
+
+   @test
+   subroutine test_Really_Really_Really_Really_Really_Really_LongLongName_max()
+      implicit none
+
+      @assertEqual(3., add(1.,2.))
+      
+   end subroutine test_Really_Really_Really_Really_Really_Really_LongLongName_max
+
+end module Test_Long_Subroutine_Name

